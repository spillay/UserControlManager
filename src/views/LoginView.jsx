import React,{Component} from 'react';
import PropTypes from 'prop-types';
import AuthService from '../service/AuthService';
import DynamicForm from '@spillay/formmanager'; 


export default class LoginView extends React.Component {
    constructor(props) {
        super(props);
        console.log("in LoginView constructor");
        this.Auth = new AuthService();
        this.state = {
            submitted: false
        }
    } // end of constructor 

    reload = () => {
        this.forceUpdate();
    }

    checkStage = () => {
        // console.log("checkStage")
        if (this.dynForm == undefined) {
            return true;
        }
        // console.log("before switch :", this.dynForm.state.stage)
        switch (this.dynForm.state.stage) {
            case "Initial":
            case "Incomplete":
                return true;
                break;
            case "Complete":
                return false;
                break;
            default:
                return true;
        }
    }

    handleSubmit = (e) => {
        e.preventDefault();
        if (this.dynForm.checkValidations() == true) {
            var data = this.dynForm.getData();
            this.setState({ submitted: true })
            console.log("now you can submit...", data);  // alert("now you can submit..."+JSON.stringify(data))
            this.Auth.login(data).then((res) => {
                console.log(res);
                this.props.submitFunc(res);
            }).catch((error) => {
                console.log(`error :${error}`)
              })
        }
    }
    render() {
        return (
            <div>
                <main role="main" className="container pt-7">
                    <div className="row">
                        <div className="col-md-6 offset-md-4">
                            <div className="card" style={{ width: '30rem' }}>
                                <img className="card-img-top custom-image mx-auto img-thumbnail" alt="South Africa" />
                                <div className="card-body">
                                    <span><i className="fa fa fa-forward" aria-hidden="true"></i>  Login </span>
                                    <form onSubmit={this.handleSubmit}>
                                        <DynamicForm           // configure the form  controls
                                            model={this.props.modelForm}
                                            groups={this.props.groups} // groups will be 1 to 4 only 1=col-md-12,  2= col-md-6 , 3=col-md-4  4= col-md-3
                                            columns={this.props.columns}
                                            ref={(node) => this.dynForm = node}
                                            reload={this.reload} >
                                        </DynamicForm>
                                        <hr />
                                        <div className='row'>
                                            <div className='col-md-12 '>
                                                <button className="btn btn-primary btn-block" disabled={this.checkStage()} >
                                                    {(this.state.submitted) ?
                                                        <div><i className="fa fa-spinner fa-spin"></i> {'Processing...'}</div> :
                                                        <i className="fa fa fa-sign-in" aria-hidden="true"></i>
                                                    }
                                                </button>
                                            </div>
                                        </div>
                                    </form>

                                    {this.props.children}


                                    <hr />
                                </div>
                            </div>
                        </div>
                    </div>

                </main>
            </div>

        )
    }

}
LoginView.propTypes = {
    groups:PropTypes.number.isRequired,
    columns:PropTypes.string.isRequired,
    modelForm: PropTypes.string.isRequired,
    // submitFunc: PropTypes.func.isRequired
};
/*

*/
/*
 <div className='row'>
                                        <div className='col-md-12'>
                                            <Link to="/register" className="btn btn-success btn-block top-buffer" >
                                                <i className="fa fa fa-user-plus" aria-hidden="true"></i> New User</Link>

                                        </div>
                                    </div>

                                    <div className='row'>
                                        <div className='col-md-12'>
                                            <Link to="/forgotpassword" className="btn btn-secondary btn-block top-buffer" >
                                                <i className="fa fa fa-key" aria-hidden="true"></i> Forgot Password ?</Link>

                                        </div>
                                    </div>

*/
