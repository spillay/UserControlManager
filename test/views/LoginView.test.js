import React from 'react';
import LoginView from '../../src/views/LoginView.jsx'
import renderer from 'react-test-renderer';
//import logo from '../../src/images/logo';

test('LoginView test', () => {
  const loginForm ="[\
      { key: 'username', label: 'User Name (email address)', type: 'email', fieldValidations: { required: true,emailFormat: true },options: {autoComplete: 'email',placeholder: 'Email address...',icon: 'fa fa fa-envelope-o'}  },\
      { key: 'password', label: 'Password', type: 'password', fieldValidations: {required: true,minLength: 6},options: {autoComplete: 'current-password',placeholder: 'Password..',icon: 'fa fa fa-lock'}  }\
  ]";
  const component = renderer.create(
    <LoginView modelForm={loginForm} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});