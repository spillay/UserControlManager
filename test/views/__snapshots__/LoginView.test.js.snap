// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView test 1`] = `
<div>
  <main
    className="container pt-7"
    role="main"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 offset-md-4"
      >
        <div
          className="card"
          style={
            Object {
              "width": "30rem",
            }
          }
        >
          <img
            alt="South Africa"
            className="card-img-top custom-image mx-auto img-thumbnail"
          />
          <div
            className="card-body"
          >
            <span>
              <i
                aria-hidden="true"
                className="fa fa fa-forward"
              />
                Login 
            </span>
            <form
              onSubmit={[Function]}
            >
              <hr />
              <div
                className="row"
              >
                <div
                  className="col-md-12 "
                >
                  <button
                    className="btn btn-primary btn-block"
                    disabled={true}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa fa-sign-in"
                    />
                  </button>
                </div>
              </div>
            </form>
            <hr />
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
